server/models/pattern/core/pattern.go

```diff
@@ -317,6 +317,9 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, cal
 		if err := json.Unmarshal(svcByt, &svc); err != nil {
 			return fmt.Errorf("failed to create service from the metadata in the scratch")
 		}
+		if svc.Name == "" {
+			return fmt.Errorf("cannot save service with empty name")
```

Comment:
What Meshery error code belongs to this error? Suggested remediation?

Reply:
This internal error is returned and wrapped into a meshklt error where the function is called. `ErrSavePattern`

Reply:
Good deal. Although - https://docs.meshery.io/search/?q=ErrSavePattern. ☹️

Reply:
I see "ErrSavePatternCode", but its description is not helpful in this situation. https://docs.meshery.io/reference/error-codes

Reply:
@acald-creator We need to continue our audit of error codes.